src/Database.cpp

```diff
@@ -1,424 +1,356 @@
-#include "contrib/sqlite/sqlite3pp.h"
-#include "contrib/bcrypt/BCrypt.hpp"
-#include "CNProtocol.hpp"
 #include "Database.hpp"
+#include "contrib/bcrypt/BCrypt.hpp">
+#include "CNProtocol.hpp"
+#include <string>
+#include "contrib/JSON.hpp"
 #include "CNStructs.hpp"
 #include "settings.hpp"
 #include "Player.hpp"
-#include <regex>
-#include <fstream>
-#include "contrib/JSON.hpp"
-
-
-//TODO: replace this sqlite wrapper with something better, clean up queries, get rid of json
-
-sqlite3pp::database db;
-
-void Database::open() {
-    //checking if database file exists
-    std::ifstream file;
-    file.open("data.db");
-
-    if (file) {
-        file.close();
-        // if exists, assign it		
-        db = sqlite3pp::database("data.db");
-        DEBUGLOG(std::cout << "[DB] Database in operation" << std::endl;)
-    }
-    else {
-        // if doesn't, create all the tables
-        DEBUGLOG(std::cout << "[DB] Creating new database" << std::endl;)
-        db = sqlite3pp::database("data.db");
-
-        // creates accounts
-        db.execute("CREATE TABLE Accounts(AccountID INTEGER PRIMARY KEY AUTOINCREMENT, Login TEXT NOT NULL, Password TEXT NOT NULL);");
-        // creates characters
-        db.execute("CREATE TABLE Players(PlayerID INTEGER PRIMARY KEY AUTOINCREMENT, AccountID INTEGER NOT NULL, Slot INTEGER NOT NULL, FirstName TEXT NOT NULL, LastName TEXT NOT NULL, CharData TEXT NOT NULL);");
-
-        DEBUGLOG(std::cout << "Done" << std::endl;)
-    }
+#include "CNStructs.hpp"
+#include "contrib/sqlite/sqlite_orm.h"
+
+using namespace sqlite_orm;
+
+# pragma region DatabaseScheme
+auto db = make_storage("database.db",
+	make_table("Accounts",
```

Comment:
Looks like the file was indented with tabs, can you change that to 4 spaces per tab instead?

Reply:
fixed, and changed my editor settings for the future